.buildconfig-android.yml

```diff
@@ -43,6 +43,20 @@ projects:
     - name: places
       type: aar
     description: Low level places storage implementation.
+  rsclient:
+    path: components/support/rs_client/android
+    artifactId: rsclient
+    publications:
+    - name: rsclient
+      type: aar
+    description: A Remote Settings client
+  remotesettings:
+    path: components/remote_settings/android
+    artifactId: remotesettings
+    publications:
+    - name: remotesettings
+      type: aar
+    description: A Remote Settings client
```

Comment:
Another idea: keep `remotesettings` UniFFIed, but don't UniFFI `rs_client`.

It'll be more boilerplate-y for sure, but I think you could still make that work, by having the UniFFIed `remotesettings` types implement `From<rs_client::{Type}>`. So instead of using externs, `remotesettings::Attachment` (for example) could `impl From<rs_client::Attachment> for remotesettings::Attachment`, and so on.

Then we can explain the split as: "`rs_client` is the low-level Remote Settings HTTP client for other Rust code only, and `remotesettings` is the high-level API for other consumers."

But I'm not sure how I feel about doing it this way 👀 WDYT?

Reply:
>> Another idea: keep `remotesettings` UniFFIed, but don't UniFFI `rs_client`.
>
> I'm not sure I agree with that exactly - in other places we are ending with with crates being uniffi'd just for their types - eg, I just landed a sync15.udl with just types because ending up with duplicate struct definitions across different crates quickly became annoying.

Oooh, thanks, this is helpful to know! To be honest, I was already feeling pretty lukewarm about suggesting this idea already, but now I think I'm also against it—for all the reasons you mentioned, for the increase in boilerplate, and also because consolidating into a single crate / component is a better way forward in every way! 😄

> That said, I'm wondering if we would see more benefit from consolidating into a single crate/component. It seems like it could continue to be confusing to future consumers at the Rust layer whether they should use `rs_client` or `remote_settings`, and I don't immediately see any reason that methods exposed to the application layer couldn't also be exposed to the Rust layer.

Yeah, I agree, too! We can have one crate and one UDL file that exposes the types (that are currently in the `rs_client` crate) and the `RemoteSettings` interface (that's currently in the `remotesettings` crate).

As for `rs_client::Client`, I think we can either have it in the single crate but not UniFFIed (so only other Rust code can use it), or expose it in UDL as a lower-level `Client` or `HttpClient` interface.

WDYT @MatthewTighe @mhammond?

Reply:
> WDYT @MatthewTighe @mhammond?

To make the "what" a little more concrete, I'm suggesting that we:

1. Choose which crate we want to use: `rs_client` or `remotesettings`.
2. Move everything from the crate that we didn't choose to the one we chose. If we chose `remotesettings`, we'd move `RemoteSettingsConfig`, `RemoteSettingsResponse`, `Record`, `Attachment`, `Client`, etc. out of `rs_client` and into `remotesettings`. If we chose `rsclient`, we'd move `RemoteSettings`, `RemoteSettingsError`, etc. out of `remotesettings` and into `rs_client`.
3. Consolidate the contents of `remotesettings.udl` and `rs_client.udl` in the crate we chose.
4. (Optional: decide if we want to expose `Client` in UDL. If not, we can still keep it a `pub` type that's exposed to Rust code, but not foreign code).
5. Delete the crate that we didn't choose.

How does that sound? Are there reasons we wouldn't want to do this, and instead keep the two-crate `rs_client` / `remotesettings` setup we have now?

Reply:
That sounds perfect to me. However, I thought the idea was that nimbus would reference rs_client. So you are suggesting that `remote_settings` could be *both* a "top level" component (ie, exposed directly to the mobile clients) *and* a dependency of nimbus? If that works I'd be fine with it - less is more!

Reply:
> So you are suggesting that `remote_settings` could be *both* a "top level" component (ie, exposed directly to the mobile clients) *and* a dependency of nimbus?

Yes, exactly!

Reply:
Awesome, glad this was resolved while I was out last week. I'm in agreement with this plan as well!